packages/components/src/tab-panel/index.tsx

```diff
@@ -110,6 +111,12 @@ export function TabPanel( {
 		}
 	}, [ tabs, selectedTab?.name, initialTabName, handleTabSelection ] );
 
+	useEffect( () => {
+		if ( selectedTabName ) {
+			setSelected( selectedTabName );
```

Comment:
> I think the convention is that the onSelect() should only be triggered from a user interaction on a tab button, not whenever the provided value changes

I think this is taken care. Calling `setSelected` updates the `selectedTab` object, and doesn't call `handleTabSelection` so it avoids the `onSelect` call. Can you confirm if it is still an issue.

> The code is a bit hard to follow at the moment because of the unclear naming between selectedTabName and selectedTab.name

Agree. But to maintain the consistency with the other prop `initialTabName` I had to use `selectedTabName`
I skipped updating `selectedTab` variable as it represents tab object and `selectedTabName` is a string.
Please suggest if we should rename any of them or add some comments for better readability.

Reply:
> I think this is taken care. Calling `setSelected` updates the `selectedTab` object, and doesn't call `handleTabSelection` so it avoids the `onSelect` call. Can you confirm if it is still an issue.

Yes, it looks good now 👍  I don't remember what it was before, but maybe I misread!

> Agree. But to maintain the consistency with the other prop `initialTabName` I had to use `selectedTabName` I skipped updating `selectedTab` variable as it represents tab object and `selectedTabName` is a string. Please suggest if we should rename any of them or add some comments for better readability.

One way I've seen it being handled in our codebase is to immediately rename it at destructuring time and append `Prop`. So in this case that would be to destructure our prop as `selectedTabName: selectedTabNameProp`. (Search the codebase for `valueProp` for some examples of this pattern to distinguish a prop value from an internal state value.)